src/domain/model/user/User.ts

```diff
@@ -1,8 +1,12 @@
-import { v4String } from 'uuid/interfaces';
 import { Email } from './Email';
+import { UserId } from './UserId';
 
 export class User {
-  constructor(private _id: string, private _email: Email) {}
+  constructor(
+    private _id: UserId,
+    private _email: Email,
+    private _creationTime: Date = new Date()
+  ) {}
 
   public id() {
```

Comment:
You could also do:
```
class Thing {
  public readonly email: Email;
}
```
And the type system will enforce immutability

Reply:
You could also use straight-up getters:

```
public get email() {
  return this._email;
}
```

Then you can just do `user.email`

Reply:
hmmm nice.. didn't know of this..